Jack/Property.fs

```diff
@@ -2,83 +2,153 @@
 
 open FSharpx.Collections
 
-type Result =
-    | Failure of List<string>
-    | Success
+type Report =
+    | Report of List<string>
+
+type Result<'a> =
+    | Failure
     | Discard
+    | Success of 'a
+
+type Property<'a> =
+    | Property of Gen<Report * Result<'a>>
+
+module Tuple =
```

Comment:
How about marking `Tuple` as `private`? I don't think we care about exposing it to the users. :wink: